### PR TITLE
Swagger ui redesign/font changes

### PR DESCRIPTION
**Pull request by lshaw-sb**
This PR implements the first stage of font updates for the swagger-ui facelift.

### Description
The new font design is based on 5 font families. Therefore I have decided to create mixins for these as was the approach in `_types.scss`  and also make all values variables to support a future ability to customise.  Its applied across` _layout.scss` and `_information.scss` in this request to give an example. Will complete rollout if approach approved.
The font file paths may be variables in subsequent PR to support any potential customising.
It also includes the introduction of EM as the unit of measurement for font size (as discussed with @kyle.shokey). 
Note change to `_main.scss` to increase specificity to `section.swagger-ui` was needed to have the font size percentage applied appropriately as `swagger-ui` classname exists in two places.

### Motivation and Context
swagger-ui facelift!!

### How Has This Been Tested?
`npm run test` passed

### Screenshots (if appropriate):
new fonts

### My PR contains... 

- [ ] No code changes (`src/` is unmodified: changes to documentation, CI, metadata, etc.)
- [ ] Dependency changes (any modification to dependencies in `package.json`)
- [ ] Bug fixes (non-breaking change which fixes an issue)
- [x ] Improvements (misc. changes to existing features)
- [ ] Features (non-breaking change which adds functionality)

### My changes...
- [ ] are breaking changes to a public API (config options, System API, major UI change, etc).
- [ ] are breaking changes to a private API (Redux, component props, utility functions, etc.).
- [ ] are breaking changes to a developer API (npm script behavior changes, new dev system dependencies, etc).
- [x ] are not breaking changes.

### Documentation
- [ x] My changes do not require a change to the project documentation.
- [ ] My changes require a change to the project documentation.
- [ ] If yes to above: I have updated the documentation accordingly.

### Automated tests
- [ x] My changes can not or do not need to be tested.
- [ ] My changes can and should be tested by unit and/or integration tests.
- [ ] If yes to above: I have added tests to cover my changes.
- [ ] If yes to above: I have taken care to cover edge cases in my tests.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by shockey** [comment]
## Page size reports

for production assets, using [this Chrome extension](https://chrome.google.com/webstore/detail/page-size-inspector/oepnndnpjiahgkljgbohnnccmokgcoln?hl=en-US)

Summary: **+250kB total**. Of that, 230kB is coming from new font requests, 20kB from stylesheets, negligible amounts elsewhere.

#### Before...
```
URL: file:///Users/kyle/Code/ui/dist/index.html
Mon May 13 2019 14:34:50 GMT-0700 (Pacific Daylight Time)

REQUEST                                       REQ     BYTES CACHEREQ CACHEBYTES

TOTAL___________________________________________4_1,427,165____________________

Document________________________________________1_____1,424____________________
-0 file:///Users/kyle/Code/ui/dist/index.html         1,424                    

Script__________________________________________2_1,271,735____________________
-0 file:///Users/kyle/Code/u.../swagger-ui-bun      944,369                    
-0 file:///Users/.../swagger-ui-standalone-pre      327,366                    

Stylesheet______________________________________1___154,006____________________
-0 file:///Users/kyle/Code/ui/dist.../swagger-      154,006                    

Image__________________________________________________________________________

XHR____________________________________________________________________________

Font___________________________________________________________________________

Other__________________________________________________________________________

```

#### After...

```
URL: file:///Users/kyle/Code/ui/dist/index.html
Mon May 13 2019 14:38:35 GMT-0700 (Pacific Daylight Time)

REQUEST                                       REQ     BYTES CACHEREQ CACHEBYTES

TOTAL__________________________________________14_1,678,058____________________

Document________________________________________1_____2,300____________________
-0 file:///Users/kyle/Code/ui/dist/index.html         2,300                    

Script__________________________________________2_1,271,723____________________
-0 file:///Users/kyle/Code/u.../swagger-ui-bun      944,357                    
-0 file:///Users/.../swagger-ui-standalone-pre      327,366                    

Stylesheet______________________________________1___173,799____________________
-0 file:///Users/kyle/Code/ui/dist.../swagger-      173,799                    

Image__________________________________________________________________________

XHR____________________________________________________________________________

Font___________________________________________10___230,236____________________
-0 file:///Users/kyle/.../roboto-light-webfont       19,576                    
-0 file:///Users/kyle/C.../roboto-light-webfon       25,756                    
-0 file:///Users/kyle.../roboto-medium-webfont       20,004                    
-0 file:///Users/kyle/.../roboto-medium-webfon       26,380                    
-0 file:///Users/kyl.../roboto-regular-webfont       19,308                    
-0 file:///Users/kyle.../roboto-regular-webfon       26,108                    
-0 file:///Users/kyle/C.../roboto-bold-webfont       20,040                    
-0 file:///Users/kyle/Co.../roboto-bold-webfon       26,528                    
-0 file:///Users.../robotomono-regular-webfont       20,084                    
-0 file:///Users/.../robotomono-regular-webfon       26,452                    

Other__________________________________________________________________________

```

File: src/style/_fonts.scss [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    font-family: 'Roboto-Medium';
+    font-display: swap;
+    src: url('./fonts/roboto-medium-webfont.woff2') format('woff2'),
+    url('./fonts/roboto-medium-webfont.woff') format('woff')
+}
+
+@font-face{
+    font-family: 'Roboto-Bold';
+    font-display: swap;
+    src: url('./fonts/roboto-bold-webfont.woff2') format('woff2'),
```
Comment (shockey) [inline src/style/_fonts.scss]: **Change request**: can you set the output directory for the font assets to be `dist/fonts/`? 

While keeping `dist` flat is an admirable goal, the font assets clutter the folder 😄

File: src/style/_variables.scss [hunk 2]
```diff
@@ -232,3 +241,29 @@ $topbar-download-url-button-font-color: $white !default;
 $text-body-default-font-color: $bright-gray !default;
 $text-code-default-font-color: $bright-gray !default;
 $text-headline-default-font-color: $bright-gray !default;
+
+// Fonts
+$special-text-font-family:'Roboto-Medium';
```
Comment (shockey) [inline src/style/_variables.scss]: I don't feel great about using `special` as a variable name unless it really nails what's going on.

**Question**: what should "special" mean to me in this context? Any ideas on alternative names?

File: dist/index.html [hunk 3]
```diff
@@ -7,6 +7,16 @@
     <link rel="stylesheet" type="text/css" href="./swagger-ui.css" >
     <link rel="icon" type="image/png" href="./favicon-32x32.png" sizes="32x32" />
     <link rel="icon" type="image/png" href="./favicon-16x16.png" sizes="16x16" />
+    <link rel="preload" as="font" href="roboto-light-webfont.woff2" type="font/woff2">
```
[the remark this answers is not in the document]
Comment (lshaw-sb) [reply dist/index.html]: I have removed the links completely as per suggestion below

File: src/style/_fonts.scss (hunk 1 above)
Comment (lshaw-sb) [reply src/style/_fonts.scss]: done

File: src/style/_variables.scss (hunk 2 above)
Comment (lshaw-sb) [reply src/style/_variables.scss]: renamed to `alt`

File: src/style/_fonts.scss [hunk 4]
```diff
@@ -0,0 +1,34 @@
+@font-face{
```
[the remark this answers is not in the document]
Comment (lshaw-sb) [reply src/style/_fonts.scss]: removed `link` . `url` works fine alone and only loads `woff2` if its available. It will only attempt to load  `woff` if it cant load `woff2`

**Comment by shockey** [comment]
thanks @lshaw-sb!
